### ISSUE TITLE
index out of bounds error in combine_1d for NIRISS WFSS

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#4987]
_Issue [JP-1473](https://jira.stsci.edu/browse/JP-1473) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

The calwebb_spec3 step combine_1d is giving the following error when processing a set of NIRISS WFSS data:
```java
2020-05-22 13:52:45,094 - stpipe.Spec3Pipeline.extract_1d - INFO - Step extract_1d done
2020-05-22 13:53:33,899 - stpipe.Spec3Pipeline.combine_1d - INFO - Step combine_1d running with args (<MultiSpecModel from jw00625-o030_s00202_niriss_f090w-gr150c-gr150r_x1d.fits>,).
2020-05-22 13:53:33,900 - stpipe.Spec3Pipeline.combine_1d - INFO - Step combine_1d parameters are: {'pre_hooks': [], 'post_hooks': [], 'output_file': None, 'output_dir': None, 'output_ext': '.fits', 'output_use_model': False, 'output_use_index': True, 'save_results': True, 'skip': False, 'suffix': 'c1d', 'search_output_file': True, 'input_dir': '/ifs/archive/test/jwst/store/doggett/tests/nOPS6/dtest2/JWSTDP-2020.2.0-65~a1ca0127/2020-05-21-214239/sdp/asn_creation/cal/level3', 'exptime_key': 'exposure_time'}
2020-05-22 13:53:33,913 - stpipe.Spec3Pipeline.combine_1d - INFO - Using exposure time as the weight.
2020-05-22 13:53:33,938 - stpipe.Spec3Pipeline.combine_1d - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/scipy/interpolate/interpolate.py:2537: RuntimeWarning: invalid value encountered in true_divide
  (grid[i + 1] - grid[i]))


Traceback (most recent call last):
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/bin/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 156, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/stpipe/cmdline.py", line 346, in step_from_cmdline
    step.run(*positional)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/pipeline/calwebb_spec3.py", line 214, in process
    result = self.combine_1d(result)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/combine_1d/combine_1d_step.py", line 26, in process
    self.exptime_key)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/combine_1d/combine1d.py", line 547, in combine_1d_spectra
    output_spec.assign_wavelengths(input_spectra)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/combine_1d/combine1d.py", line 138, in assign_wavelengths
    (wl, n_input_spectra) = count_input(input_spectra)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.1.20200520-py37/lib/python3.7/site-packages/jwst/combine_1d/combine1d.py", line 332, in count_input
    if input_wl[1] > input_wl[0]:       # wavelengths are increasing
IndexError: index 1 is out of bounds for axis 0 with size 1
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec3Pipeline':
    index 1 is out of bounds for axis 0 with size 1
----------------------------------------------------------------------
```
Not sure if this is a garbage data related problem.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) on [JIRA](https://jira.stsci.edu/browse/JP-1473?focusedCommentId=468581#comment-468581):_

Though still possibly a "garbage-in" issue, there are many slits that are very small, less than 10 pixels.  The error is occurring at a point where only overlapping wavelengths are being considered, easily causing a situation of only a single pixel overlap. Below is a count of slit size vs how many of that size occur. A surprise to myself is how little slit length consistency there is. My presumption would be that slit length would be much more consistent.

Will investigate a bit more, but in light of jp-1385 and having random input data, not sure what will be resolved here.

 {2: 1,
 3: 3,
 4: 3,
 5: 2,
 6: 3,
 7: 3,
 8: 7,
 9: 65,
 10: 357,
 11: 572,
 12: 433,
 13: 195,
 14: 141,
 15: 61,
 16: 6,
 17: 22,
 18: 14,
 19: 1,
 20: 5,
 21: 1,
 22: 4,
 24: 4,
 25: 3,
 26: 2,
 27: 3,
 28: 1,
 29: 2,
 30: 3,
 31: 2,
 32: 1,
 33: 3,
 35: 11,
 36: 9,
 37: 7,
 38: 23,
 39: 53,
 40: 92,
 41: 193,
 42: 273,
 43: 241,
 44: 49,
 45: 2,
 46: 1,
 48: 3,
 49: 2,
 50: 2,
 52: 1,
 53: 3,
 54: 2,
 55: 2,
 57: 1,
 58: 2,
 59: 3,
 60: 1,
 61: 1,
 62: 2,
 64: 1,
 65: 3,
 66: 1,
 67: 1,
 68: 1,
 70: 1,
 76: 1,
 78: 1,
 79: 1,
 83: 1,
 85: 8,
 86: 10,
 88: 14,
 89: 48,
 90: 88,
 91: 178,
 92: 258,
 93: 125,
 94: 9}

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1473?focusedCommentId=469097#comment-469097):_

Fixed by #5037
